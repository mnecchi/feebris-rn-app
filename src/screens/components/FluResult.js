import React from 'react';
import {View, Text} from 'react-native';

const FluResult = ({
  isFlu,
  containerStyle,
  textStyle,
  textFluStyle,
  textNoFluStyle,
}) => (
  <View style={containerStyle}>
    <Text style={[textStyle, isFlu ? textFluStyle : textNoFluStyle]}>
      {isFlu ? "Sorry, you've got the Flu!" : 'No Flu! Hooray!'}
    </Text>
    <Text style={textStyle}>{isFlu ? 'ðŸ’Š' : 'ðŸŽ‰'}</Text>
  </View>
);

export default FluResult;
