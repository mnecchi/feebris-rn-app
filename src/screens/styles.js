import {StyleSheet} from 'react-native';

const labelText = {fontSize: 30, fontWeight: 'bold', color: '#444'};
const fieldContainer = {
  justifyContent: 'center',
  alignItems: 'center',
  margin: 16,
  maxWidth: '80%',
};
const switchContainer = {
  ...fieldContainer,
  textAlignVertical: 'center',
};

const iconInputContainer = {
  flexDirection: 'row',
  marginTop: 8,
  alignItems: 'center',
  justifyContent: 'center',
};

const homeStyle = StyleSheet.create({
  container: {
    flex: 1,
    height: '100%',
  },
  scrollContainer: {
    flex: 1,
    width: '100%',
  },
  scrollContent: {alignItems: 'center'},
  // TEMPERATURE
  temperatureContainer: fieldContainer,
  temperatureText: labelText,
  temperatureInput: {
    borderColor: '#ccc',
    borderWidth: 1,
    borderRadius: 8,
    padding: 8,
    fontSize: 20,
    width: 100,
    margin: 8,
  },
  temperatureInputError: {
    borderColor: 'red',
  },
  temperatureInputContainer: iconInputContainer,
  // COUGH
  coughContainer: switchContainer,
  coughText: {...labelText, paddingRight: 8},
  coughSwitchContainer: iconInputContainer,
  // FEVER IN LAST 5 DAYS
  feverContainer: switchContainer,
  feverText: {...labelText, paddingRight: 8},
  feverSwitchContainer: iconInputContainer,
  // BUTTON
  buttonContainer: {
    padding: 8,
    backgroundColor: 'rgba(60, 142, 183, 1)',
    borderRadius: 8,
    marginTop: 64,
    width: '80%',
    maxWidth: 600,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonText: {...labelText, color: 'white'},
});

const resultsStyle = StyleSheet.create({
  container: {flex: 1, flexDirection: 'column'},
  fluResultContainer: {
    alignItems: 'center',
    marginTop: 32,
    marginHorizontal: 16,
    paddingBottom: 32,
    borderBottomColor: '#ccc',
    borderBottomWidth: 1,
  },
  fluResultText: {
    ...labelText,
  },
  fluResultFlu: {
    color: 'red',
  },
  fluResultNoFlu: {
    color: 'green',
  },
  readingsTable: {
    borderWidth: 1,
    borderColor: '#ccc',
  },
  readingsTableHeader: {
    backgroundColor: 'rgba(60, 142, 183, 0.5)',
  },
  readingTableCell: {
    margin: 8,
    textAlign: 'center',
  },
  tableContainer: {
    marginTop: 16,
    marginHorizontal: 8,
  },
});

export {homeStyle, resultsStyle};
