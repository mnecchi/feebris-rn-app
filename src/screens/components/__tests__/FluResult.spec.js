import React from 'react';
import 'react-native';
import FluResult from '../FluResult';
import renderer from 'react-test-renderer';
import {resultsStyle} from '../../styles';

describe('should render successfully', () => {
  it('is flu', () => {
    const tree = renderer
      .create(
        <FluResult
          isFlu={true}
          containerStyle={resultsStyle.fluResultContainer}
          textStyle={resultsStyle.fluResultText}
          textFluStyle={resultsStyle.fluResultFlu}
          textNoFluStyle={resultsStyle.fluResultNoFlu}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('is not flu', () => {
    const tree = renderer
      .create(
        <FluResult
          isFlu={false}
          containerStyle={resultsStyle.fluResultContainer}
          textStyle={resultsStyle.fluResultText}
          textFluStyle={resultsStyle.fluResultFlu}
          textNoFluStyle={resultsStyle.fluResultNoFlu}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
