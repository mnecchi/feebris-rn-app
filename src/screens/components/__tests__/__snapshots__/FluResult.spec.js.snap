// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render successfully is flu 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginHorizontal": 16,
      "marginTop": 32,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#444",
          "fontSize": 30,
          "fontWeight": "bold",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Sorry, you've got the Flu!
  </Text>
  <Text
    style={
      Object {
        "color": "#444",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    ðŸ’Š
  </Text>
</View>
`;

exports[`should render successfully is not flu 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginHorizontal": 16,
      "marginTop": 32,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#444",
          "fontSize": 30,
          "fontWeight": "bold",
        },
        Object {
          "color": "green",
        },
      ]
    }
  >
    No Flu! Hooray!
  </Text>
  <Text
    style={
      Object {
        "color": "#444",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    ðŸŽ‰
  </Text>
</View>
`;
